packages/amplify-e2e-tests/src/rds-v2-tests-common/rds-auth-userpool-fields.ts

```diff
@@ -0,0 +1,1925 @@
+      const createResult = await todoHelper.create(createResultSetName, todoWithPrivateFields, privateResultSet);
+      expect(createResult.data[createResultSetName].id).toBeDefined();
+      todo['id'] = createResult.data[createResultSetName].id;
+
+      const todoUpdated = {
+        id: todo['id'],
+        privateContent: 'Private Content updated',
+        publicContent: 'Public Content updated',
+      };
+
+      await expect(async () => await todoHelper.update(`update${modelName}`, todoUpdated)).rejects.toThrowErrorMatchingInlineSnapshot(
+        `"GraphQL error: Not Authorized to access update${modelName} on type Mutation"`,
+      );
+
+      const expectedFieldError = `"GraphQL error: Not Authorized to access publicContent on type ${modelName}"`;
+      const getResult = await todoHelper.get({ id: todo['id'] }, undefined, true, 'all');
+      checkOperationResult(getResult, { ...todo, publicContent: null }, `get${modelName}`, false, [expectedFieldError]);
+
+      const listTodosResult = await todoHelper.list({}, completeResultSet, `list${modelName}`, false, 'all');
+      checkListItemExistence(listTodosResult, `list${modelName}`, todo['id'], true);
+      checkListResponseErrors(listTodosResult, [`Not Authorized to access publicContent on type ${modelName}`]);
+
+      const todoRandom = {
+        ...todoWithPrivateFields,
+        id: Date.now().toString(),
+      };
+      const todoRandomUpdated = {
+        ...todoWithPrivateFields,
+        id: todoRandom.id,
+      };
+      const actorClient = getConfiguredAppsyncClientCognitoAuth(graphQlEndpoint, region, userMap[userName1]);
+      const subTodoHelper = createModelOperationHelpers(actorClient, schema)[modelName];
+
+      const onCreateSubscriptionResult = await subTodoHelper.subscribe('onCreate', [
+        async () => {
+          await subTodoHelper.create(createResultSetName, todoRandom, privateResultSet);
+        },
+      ]);
+      expect(onCreateSubscriptionResult).toHaveLength(1);
+      checkOperationResult(
+        onCreateSubscriptionResult[0],
+        { ...todoRandom, publicContent: null, privateContent: null },
+        `onCreate${modelName}`,
+      );
+
+      const onUpdateSubscriptionResult = await subTodoHelper.subscribe('onUpdate', [
+        async () => {
+          await subTodoHelper.update(`update${modelName}`, todoRandomUpdated, privateResultSet);
+        },
+      ]);
+      expect(onUpdateSubscriptionResult).toHaveLength(1);
+      checkOperationResult(
+        onUpdateSubscriptionResult[0],
+        { ...todoRandomUpdated, publicContent: null, privateContent: null },
+        `onUpdate${modelName}`,
+      );
+
+      const onDeleteSubscriptionResult = await subTodoHelper.subscribe('onDelete', []);
+      expect(onDeleteSubscriptionResult).toHaveLength(0);
+    });
+
+    test.skip('user not part of allowed custom group of a record can perform CRUD and subscription operations with only allowed fields in selection set', async () => {
```

Comment:
As above, add an easily searchable todo, or link to this code in our backlog ticket